### PR TITLE
dashboard refactor

### PR DESCRIPTION
**Pull request by kiwis08**
## Summary
Huge refactor:

- new folder structure 
- dashboard screen was refactored to achieve a better architecture
- several folders were included: services, queries, provider, models, queries, etc.
- Graphql widgets are no longer used
- User information is stored in provider
- dashboard screen widget was cleaned,  most of the logic is now handle in the controller

## Considerations
- There is an error when going to the user profile tab
- Other screens may be broken. I suggest we fix these things in other PRs

## Type of change
- [ ] feature
- [ ] hotfix
- [ ] fix
- [x] refactor
- [ ] chore
- [ ] docs

## How to test?
Login, wait for dashboard to load and check user info is correctly updated, go to user profile where username should show as well as the transactions, return to dashboard for user info to fetch again.

## How has this been tested?
- [ ] Toolbar should show username after dashboard loads
- [ ] User profit should be 0 and balance should load from backend
- [ ] User info should update when returning to dashboard
- [ ] Dashboard should transactions. If there aren't any, it should a corresponding message

## Checklist (for the reviewer)

- [ ] Does the code compile without errors or warnings?
- [ ] Is the PR free of merge conflicts?
- [ ] Is the UI responsive?
- [ ] Is the PR modular?

### CODE REVIEW

File: lib/src/features/dashboard/data/dashboard_controller.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  Future<List<Transaction>?> _getUserTransactions() async {
+    StorageService service = StorageService();
+    var token = await service.readSecureData("token");
+    GraphQLConfig graphQLConfiguration = GraphQLConfig(token: "${token}");
+    GraphQLClient client = graphQLConfiguration.clientToQuery();
+    var result = await client.query(QueryOptions(document: gql(transactionsQuery), variables: getUserTransactionsInput));
+    Map<String, dynamic> response =
+    result.data?['getUserTransactions'];
```
Comment (manu-yaff) [inline lib/src/features/dashboard/data/dashboard_controller.dart:27]: Creo que aquí deberíamos manejar errores en caso de que sucedan

File: lib/src/features/dashboard/data/dashboard_controller.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    return transactionsList;
+  }
+
+  Future<double> _getUserProfit() async {
+    StorageService service = StorageService();
+    var token = await service.readSecureData("token");
+    GraphQLConfig graphQLConfiguration = GraphQLConfig(token: "${token}");
+    GraphQLClient client = graphQLConfiguration.clientToQuery();
+    var result = await client.query(QueryOptions(document: gql(userProfitQuery)));
+    double response = result.data?['getUserProfit'];
```
Comment (manu-yaff) [inline lib/src/features/dashboard/data/dashboard_controller.dart:40]: Creo que aquí deberíamos manejar errores en caso de que sucedan

File: lib/src/features/dashboard/data/dashboard_controller.dart [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  Future<User> _getUser() async {
+    StorageService service = StorageService();
+    var token = await service.readSecureData("token");
+    GraphQLConfig graphQLConfiguration = GraphQLConfig(token: "${token}");
+    GraphQLClient client = graphQLConfiguration.clientToQuery();
+    var result = await client.query(QueryOptions(document: gql(meQuery)));
+
+    if (result.hasException) {
+      print("Exception");
```
Comment (manu-yaff) [inline lib/src/features/dashboard/data/dashboard_controller.dart:52]: Quitemos este "print" jaja

File: lib/src/features/dashboard/data/dashboard_controller.dart [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      final e = result.exception?.graphqlErrors[0].message;
+      return Future.error(e!);
+    } else if (result.data != null) {
+      final data = result.data!["me"] as Map<String, dynamic>;
```
Comment (manu-yaff) [inline lib/src/features/dashboard/data/dashboard_controller.dart:56]: Así como haces estás validaciones sería hacer lo mismo en las funciones de arriba

File: lib/src/features/dashboard/data/dashboard_controller.dart [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      return user;
+    }
+    return Future.error("Error getting user information");
+    // print(result.hasException);
+  }
+
+  Future<User> getAllUserInformation() async {
+    try {
+      final user = await _getUser();
+      //final userProfit = await _getUserProfit();
```
Comment (manu-yaff) [inline lib/src/features/dashboard/data/dashboard_controller.dart:67]: Quitemos estos comentarios

File: lib/src/features/dashboard/data/dashboard_controller.dart [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    if (result.hasException) {
+      print("Exception");
+      final e = result.exception?.graphqlErrors[0].message;
+      return Future.error(e!);
+    } else if (result.data != null) {
+      final data = result.data!["me"] as Map<String, dynamic>;
+      final user = User.fromJSON(data);
+      return user;
+    }
+    return Future.error("Error getting user information");
```
Comment (manu-yaff) [inline lib/src/features/dashboard/data/dashboard_controller.dart:60]: Tengo duda cuál es la mejor manera de regresar los errores, por qué así y no un Throw Exception()?

File: lib/src/features/dashboard/presentation/dashboard_screen.dart [hunk 7]
```diff
@@ -1,133 +1,75 @@
...
-                      response,
-                    ).getTransactions();
-
-                    return Container(
-                      padding: const EdgeInsets.only(
-                        left: 24,
-                        right: 24,
-                        bottom: 24,
-                        top: 0,
+    final userInfo = ref.watch(userInformationProvider);
```
Comment (manu-yaff) [inline lib/src/features/dashboard/presentation/dashboard_screen.dart:16]: Ya se ve mucho más limpio sin tener los widgets de Query, bien bien

File: lib/src/features/dashboard/presentation/dashboard_screen.dart [hunk 8]
```diff
@@ -1,133 +1,75 @@
...
             ),
-          );
-        }
-        return const Center(child: CircularProgressIndicator());
-      },
-    );
+          ),
+        ),
+      );
+    }, error: (error, stack) {
```
Comment (manu-yaff) [inline lib/src/features/dashboard/presentation/dashboard_screen.dart:69]: Creo que cuando sucede un error en back, pero el response es ok, aquí no va a cachear el error

File: lib/src/features/dashboard/data/dashboard_controller.dart [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    } catch (e) {
+      return Future.error(e);
+    }
+  }
+}
+
+
+
+final dashboardControllerProvider = Provider((ref) => DashboardController());
+final userInformationProvider = FutureProvider.autoDispose<User>((ref) {
```
Comment (juandatorr99) [inline lib/src/features/dashboard/data/dashboard_controller.dart:81]: La idea seria que el userprovider se guarde y que si el usuario quiere recargar haya un swipe down to load

**Pull request merged**